examples/vector_databases/timescale/time_based_filtering_with_langchain.ipynb

```diff
@@ -0,0 +1,455 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Similarity Search with time-based filtering using LangChain\n",
+    "\n",
+    "A key use case for Timescale Vector is efficient time-based vector search. Timescale Vector enables this by automatically partitioning vectors (and associated metadata) by time. This allows you to efficiently query vectors by both similarity to a query vector and time because we can exclude entire partitions that don't overlap with the query time.\n",
+    "\n",
+    "Time-based vector search functionality is helpful for applications like:\n",
+    "- Storing and retrieving LLM response history (e.g. chatbots)\n",
+    "- Finding the most recent embeddings that are similar to a query vector (e.g recent news).\n",
+    "- Constraining similarity search to a relevant time range (e.g asking time-based questions about a knowledge base)\n",
+    "\n",
+    "To illustrate how to use TimescaleVector's time-based vector search functionality, we'll ask questions about the git log history for TimescaleDB . We'll illustrate how to add documents with a time-based uuid and how run similarity searches with time range filters.\n",
+    "\n",
+    "## Setup your environment\n",
+    "\n",
+    "First, install the necessary prerequisites"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Pip install necessary packages\n",
+    "%pip install timescale-vector\n",
+    "%pip install openai\n",
+    "%pip install langchain\n",
+    "%pip install python-dotenv"
```

Comment:
add `pip install jq` is required
I also had to add `pip install tiktoken` to get it to work